### PR TITLE
Creating a function for a new tarball URL

### PR DESCRIPTION
**Pull request by tomahawk28**
Following https://github.com/giantswarm/appcatalog/pull/11

Merely copy from https://github.com/giantswarm/app-operator/blob/master/pkg/tarball/tarball.go

### CODE REVIEW

File: appcatalog.go [hunk 1]
```diff
@@ -50,6 +52,19 @@ func GetLatestVersion(ctx context.Context, storageURL, app string) (string, erro
 	return version, nil
 }
 
+// NewTarballURL returns the tarball URL of specified chart combining with app name, version.
```
Comment (rossf7) [inline appcatalog.go]:
```suggestion
// NewTarballURL returns the chart tarball URL for the specified app and version.
```

File: error.go [hunk 2]
```diff
@@ -2,6 +2,15 @@ package appcatalog
 
 import "github.com/giantswarm/microerror"
 
+var executionFailedError = &microerror.Error{
+	Kind: "executionFailed",
+}
+
+// IsExecutionFailed asserts executionFailedError.
+func IsExecutionFailed(err error) bool {
```
Comment (xh3b4sd) [inline error.go]: Please see https://github.com/giantswarm/fmt/blob/master/go/errors.md#matching-errors and align this with `aws-operator`.

File: appcatalog.go [hunk 3]
```diff
@@ -50,6 +52,19 @@ func GetLatestVersion(ctx context.Context, storageURL, app string) (string, erro
...
+func NewTarballURL(baseURL string, appName string, version string) (string, error) {
+	if baseURL == "" || appName == "" || version == "" {
+		return "", microerror.Maskf(executionFailedError, "baseURL %#q, appName %#q, release %#q should not be empty", baseURL, appName, version)
```
Comment (rossf7) [inline appcatalog.go]: Like Tim says we shouldn't use `executionFailedError` here. What about `invalidTarballError`?

Comment (tomahawk28) [reply appcatalog.go]: it makes sense, I will change the error name in here and matcher, too.

File: error.go [hunk 4]
```diff
@@ -2,6 +2,15 @@ package appcatalog
 
 import "github.com/giantswarm/microerror"
 
+var invalidTarballError = &microerror.Error{
+	Kind: "executionFailed",
```
Comment (rossf7) [inline error.go]:
```suggestion
	Kind: "invalidTarballError",
```

Definitely. The error needs to be aligned with `fmt` but what I meant was we need an error matcher.

**Comment by tomahawk28** [comment]
I forgot to push README changes from my previous PR, just sorted it out now. Thanks !

File: error.go [hunk 5]
```diff
@@ -2,6 +2,15 @@ package appcatalog
 
 import "github.com/giantswarm/microerror"
 
+var invalidTarballError = &microerror.Error{
```
Comment (kopiczko) [inline error.go]: Is it going to be handled somewhere? It should be `executionFailedError` otherwise.

Comment (rossf7) [reply error.go]: Hmm well if we can't generate the tarball URL then the chart resource in app-operator needs to fail.

Execution failed is a very generic error. Isn't is better to return an invalid tarball error?

Comment (tomahawk28) [reply error.go]: At least in app-operator, we need to handle this error.

Comment (kopiczko) [reply error.go]: When `NewTarballURL` fails then URL can't be generated. Do we need a specific error here?

Comment (tomahawk28) [reply error.go]: I think we don't need a specific error here as you said. In many cases when the URL failed to generate, we should cancel the resource. Is it okay to replace it as executionFailed and remove a matcher as later PR?

Comment (rossf7) [reply error.go]: OK cool as long as the error text is meaningful this is fine. Let's do that.

**Pull request merged**